packages/hasher/src/repoInfo.ts

```diff
@@ -29,24 +31,97 @@ function searchRepoInfoCache(packageRoot: string) {
   }
 }
 
+function createPackageHashes(
```

Comment:
Optional: consider moving this to it's own file and adding a test. Not required, but nice to have small testable units.